src/ray/gcs/gcs_server/gcs_actor_manager.cc

```diff
@@ -444,6 +441,20 @@ Status GcsActorManager::RegisterActor(
       [this, actor](const Status &status) {
         // The backend storage is supposed to be reliable, so the status must be ok.
         RAY_CHECK_OK(status);
+        // If a creator dies before this callback is called, the actor could have been
```

Comment:
At present, the back-end storage is Redis, `gcs_table_storage_` uses the same socket to access Redis, so I think the order can be guaranteed. If the `gcs_table_storage_->ActorTable().Put` could guarantee the order of `Put`, is this PR unnecessary?

Reply:
I see. We still need return statement here (that’s the original issue here). I also think it is probably a bad practice to assume the order is guaranteed given we might support multiple backends with abstract interface (this can lead us to disastrous result later).

Reply:
Did gcs server currently written with assumption that the order that calls backend is guaranteed?

Reply:
> I see. We still need return statement here (that’s the original issue here). I also think it is probably a bad practice to assume the order is guaranteed given we might support multiple backends with abstract interface (this can lead us to disastrous result later).

Okey，I think the state of DEAD is storing or stored somewhere if `registered_actor_it == registered_actors_ .end()`. So what you need to do here is do nothing and just return if `registered_actor_it == registered_actors_ .end()`?

Reply:
> I see. We still need return statement here (that’s the original issue here). I also think it is probably a bad practice to assume the order is guaranteed given we might support multiple backends with abstract interface (this can lead us to disastrous result later).

I think it is better to return if `registered_actor_it == registered_actors_.end()` because it means the actor is already marked as DEAD and is flushing or flushed.